src/Microsoft.AspNetCore.Mvc.TagHelpers/Internal/CacheTagHelperMemoryCacheFactory.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Mvc.TagHelpers.Internal
+{
+    public class CacheTagHelperMemoryCacheFactory
+    {
+        public static readonly string MemoryCacheOptionsName = typeof(CacheTagHelper).FullName;
+
+        public CacheTagHelperMemoryCacheFactory(IOptionsMonitor<MemoryCacheOptions> optionsMonitor)
+        {
+            Cache = new MemoryCache(optionsMonitor.Get(MemoryCacheOptionsName));
+        }
+
+        // For testing only.
+        internal CacheTagHelperMemoryCacheFactory(IMemoryCache cache)
+        {
+            Cache = cache;
+        }
+
+        public IMemoryCache Cache { get; }
```

Comment:
> Alternatively, just make the internal constructor public to allow people to register this type in DI with their own IMemoryCache instance.

Making this public means we have two 'DI' constructors. What about a static factory method? `CTHMCFactory.Create(IMemoryCache ...)`

Or you can make `CacheTagHelperMemoryCacheFactory(IMemoryCache cache)` public, delete `CacheTagHelperMemoryCacheFactory(IOptionsMonitor<MemoryCacheOptions> optionsMonitor)` and register this in DI using a factory method.